src/main/java/org/apache/sling/i18n/impl/JcrResourceBundleProvider.java

```diff
@@ -646,43 +675,95 @@ private void preloadBundles() {
      * languages and countries provided by the platform. Any unsupported
      * language or country is replaced by the platform default language and
      * country.
+     * Locale string is also parsed for script tag. Any unsupported script is ignored.
      * @param localeString the locale as string
      * @return the {@link Locale} being generated from the {@code localeString}
      */
     static Locale toLocale(String localeString) {
         if (localeString == null || localeString.length() == 0) {
             return Locale.getDefault();
         }
+
         // support BCP 47 compliant strings as well (using a different separator "-" instead of "_")
         localeString = localeString.replaceAll("-", "_");
-
         // check language and country
         final String[] parts = localeString.split("_");
         if (parts.length == 0) {
             return Locale.getDefault();
         }
 
         // at least language is available
-        String lang = parts[0];
-        boolean isValidLanguageCode = false;
-        String[] langs = Locale.getISOLanguages();
-        for (int i = 0; i < langs.length; i++) {
-            if (langs[i].equalsIgnoreCase(lang)) {
-                isValidLanguageCode = true;
-                break;
+        String lang = getValidLanguage(parts[0]);
+        if (parts.length == 1) {
+            return new Locale(lang);
+        }
+
+        Locale localeWithBuilder = createLocaleWithBuilder(parts, lang);
+        if (localeWithBuilder != null) {
+            return localeWithBuilder;
+        }
+
+        return createLocaleWithConstructor(lang, parts);
+    }
+
+    private static Locale createLocaleWithBuilder(String[] parts, String lang) {
```

Comment:
I think it would make sense for documentation purposes to add an assert statement, that ``parts.length >=2`` can be always assumed. WDYT?

Reply:
Added assert statement. Also, updated documentation part of this method.